### PR TITLE
Use a 1-element smallvec for selector lists.

### PR DESCRIPTION
**Pull request by jdm**
Profiling shows this reducing parsing time by a few milliseconds on the tp6 facebook case. The gtest benchmark with the same concatenated stylesheets also shows a significant improvement.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18395)

### CODE REVIEW

File: components/selectors/parser.rs [hunk 1]
```diff
@@ -186,13 +186,20 @@ impl<Impl: SelectorImpl> SelectorList<Impl> {
...
+                Ok(&Token::Comma) => continue,
+                Ok(_) => unreachable!(),
+            }
+        }
     }
 
     /// Creates a SelectorList from a Vec of selectors. Used in tests.
     pub fn from_vec(v: Vec<Selector<Impl>>) -> Self {
-        SelectorList(v)
+        SelectorList(v.into_iter().collect())
```
Comment (SimonSapin) [inline components/selectors/parser.rs]: `.into_vec()` reuses an existing allocation if there is one.

Comment (SimonSapin) [reply components/selectors/parser.rs]: I misread what this does. `SmallVec::from_vec`, then.

**Comment by SimonSapin** [comment]
r=me with this fixed:

```rust
error: expected one of `,` or `>`, found `;`
   --> /home/travis/build/servo/servo/components/selectors/parser.rs:179:74
    |
179 | pub struct SelectorList<Impl: SelectorImpl>(pub SmallVec<[Selector<Impl>]; 1>);
    |                                                                          ^ expected one of `,` or `>` here
error: expected item, found `1`
   --> /home/travis/build/servo/servo/components/selectors/parser.rs:179:76
    |
179 | pub struct SelectorList<Impl: SelectorImpl>(pub SmallVec<[Selector<Impl>]; 1>);
    |                                                                            ^
```

File: components/selectors/parser.rs (hunk 1 above)
Comment (emilio) [reply components/selectors/parser.rs]: Yeah, it's not a big deal, this is just for testing anyway.

**Comment by jdm** [comment]
@bors-servo: retry
* #18273

**Pull request merged**